Cargo.toml

```diff
@@ -81,6 +81,24 @@ packaging = ["watchman"]
 vendored-openssl = ["git2/vendored-openssl", "jj-lib/vendored-openssl"]
 watchman = ["jj-lib/watchman"]
 
+# It's often really useful to profile and run benchmarks on jj in --release
+# mode, at the expense of some peak performance; so bump codegen-units to a
+# high explicit number, and enable incremental builds to speed things up when
+# iterating on patches
 [profile.release]
+incremental = true # normally off in --release mode
+codegen-units = 16 # normally 256 when incremental=true; tone it down a bit...
+debug = true
+strip = "none"
```

Comment:
Oh, yeah, that's probably a showstopper, I think, since `cargo install` is a recommended method (and even if I don't particularly *like* that method, people do use it anyway.)

In theory if we actually automated the release process, there is no reason we could not strip/exchange these profiles, but... I don't know if anyone has a taste for that kind of thing.

I'll leave this up for now for a few more comments, but I might rework it as you noted or abandon this.